gravitee-apim-portal-webui-next/src/app/api-details/api-tab-subscriptions/subscriptions-details/subscriptions-details.component.spec.ts

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { HttpTestingController } from '@angular/common/http/testing';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { MatInputHarness } from '@angular/material/input/testing';
+
+import { SubscriptionsDetailsComponent } from './subscriptions-details.component';
+import { fakeApi, fakeApisResponse } from '../../../../entities/api/api.fixtures';
+import { ApisResponse } from '../../../../entities/api/apis-response';
+import { Application } from '../../../../entities/application/application';
+import { fakeApplication } from '../../../../entities/application/application.fixture';
+import { Subscription, SubscriptionData } from '../../../../entities/subscription/subscription';
+import { fakeSubscription, fakeSubscriptionResponse } from '../../../../entities/subscription/subscription.fixture';
+import { AppTestingModule, TESTING_BASE_URL } from '../../../../testing/app-testing.module';
+
+describe('SubscriptionsDetailsComponent', () => {
+  let fixture: ComponentFixture<SubscriptionsDetailsComponent>;
+  let httpTestingController: HttpTestingController;
+  let harnessLoader: HarnessLoader;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [SubscriptionsDetailsComponent, AppTestingModule],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(SubscriptionsDetailsComponent);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    harnessLoader = TestbedHarnessEnvironment.loader(fixture);
+
+    fixture.componentInstance.subscriptionApplicationId = 'testApplicationId';
+    fixture.componentInstance.apiId = 'testApiId';
+
+    fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  describe('subscription pending', () => {
+    beforeEach(() => {
+      expectSubscriptionListWithKeys(fakeSubscription({ status: 'PENDING' }));
+      expectSubscriptionList(fakeSubscriptionResponse(), 'testApiId');
+      expectApiPlansList(fakeApisResponse());
+      expectApplicationsList(fakeApplication());
+    });
+
+    it('should show pending status ', async () => {
+      expect(fixture.nativeElement.querySelector('.subscriptions-details__pending')).toBeDefined();
+    });
+  });
+
+  describe('subscription rejected', () => {
+    beforeEach(() => {
+      expectSubscriptionListWithKeys(fakeSubscription({ status: 'REJECTED' }));
+      expectSubscriptionList(fakeSubscriptionResponse(), 'testApiId');
+      expectApiPlansList(fakeApisResponse());
+      expectApplicationsList(fakeApplication());
+    });
+
+    it('should show pending status ', async () => {
+      expect(fixture.nativeElement.querySelector('.subscriptions-details__rejected')).toBeDefined();
+    });
+  });
+
+  describe('subscription accepted', () => {
```

Comment:
what do you think about adding same kind of test with PAUSED & CLOSED ?

Reply:
Done